src/v/kafka/server/handlers/fetch.cc

```diff
@@ -454,17 +471,46 @@ ss::future<read_result> read_from_ntp(
     }
 
     auto high_watermark = partition->high_watermark();
+
     auto max_offset = high_watermark < model::offset(0) ? model::offset(0)
                                                         : high_watermark;
+
+    if (config::shard_local_cfg().enable_transactions.value()) {
+        if (config.isolation_level == 1) {
```

Comment:
isolation level should be a strong type and the levels can have constexpr names predefined?